data-tck/src/main/java/io/micronaut/data/tck/repositories/SaleRepository.java

```diff
@@ -41,15 +42,15 @@ public interface SaleRepository extends CrudRepository<Sale, Long> {
     void updateData(@Id Long id, @Parameter("data") Map<String, String> data, @Parameter("dataList") List<String> dataList);
 
     @Query("SELECT extra_data AS extraData FROM sale WHERE id = :id")
-    @QueryResult(column = "extraData", type = QueryResult.Type.JSON)
+    @QueryResult(column = "extraData", type = QueryResult.Type.JSON, jsonType = JsonType.STRING)
     Optional<Discount> getDiscountById(Long id);
 
     @Query("SELECT extra_data AS DATA FROM sale WHERE name = :name")
-    @QueryResult(type = QueryResult.Type.JSON)
+    @QueryResult(type = QueryResult.Type.JSON, jsonType = JsonType.STRING)
     List<Sale> findAllByNameFromJson(String name);
 
     @Query("SELECT extra_data AS DATA FROM sale WHERE name = :name")
-    @QueryResult(type = QueryResult.Type.JSON)
+    @QueryResult(type = QueryResult.Type.JSON, jsonType = JsonType.STRING)
     Optional<Sale> findByNameFromJson(String name);
```

Comment:
I don't we need any of `@QueryResult` here, it's enough to determine the result type from the `DataType` an `JsonDataType`

Reply:
This is a bit strange test case, just used one field extra_data to read Sale from but there is no any info without `@QueryResult` how to deserialize that so I think in this case annotation is still needed.

Reply:
The custom query example is fine. We should also be able to write it as:
`Optional<Sale> findExtraDataByName(String name);`
We know that the return is a property `extraData` annotated with data type JSON

Reply:
Ok